src/track/CMakeLists.txt

```diff
@@ -12,6 +12,15 @@ install(PROGRAMS ${PROJECT_BINARY_DIR}/${BIN_INSTALL_DIR}/heaptrack
     DESTINATION ${BIN_INSTALL_DIR}
 )
 
+if (HEAPTRACK_HAVE_LIBUNWIND)
+    add_library(heaptrack_unwind STATIC trace_libunwind.cpp)
+    target_link_libraries(heaptrack_unwind LINK_PRIVATE ${LIBUNWIND_LIBRARY})
+else()
+    add_library(heaptrack_unwind STATIC trace_unwind_tables.cpp)
+endif()
+
+target_compile_options(heaptrack_unwind PRIVATE -fPIC) #some symbols (_Unwind_GetIP etc) require position independent code
```

Comment:
this should be moved into the else above

but quite frankly, I don't understand why *using* _Unwind_GetIP requires PIC. Rather, _Unwind_GetIP may not return useful data when unwinding code that wasn't compiled with -fPIC maybe? Can you please expand on this, maybe reference some documentation that says this is required?t

also, the preferred way to achieve this is via
```
set_property(TARGET lib1 PROPERTY POSITION_INDEPENDENT_CODE ON)
```

Reply:
I've learned about this from link error actually:

1. HEAPTRACK_USE_LIBUNWIND - ON

/usr/bin/ld: libheaptrack_unwind.a(trace_libunwind.cpp.o): relocation R_X86_64_32S against symbol `stderr@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Nonrepresentable section on output
clang: error: linker command failed with exit code 1 (use -v to see invocation)

1. HEAPTRACK_USE_LIBUNWIND - OFF

/usr/bin/ld: libheaptrack_unwind.a(trace_unwind_tables.cpp.o): relocation R_X86_64_PC32 against symbol `_Unwind_GetIP@@GCC_3.0' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
clang: error: linker command failed with exit code 1 (use -v to see invocation)

I believe it is necessary for building STATIC library with those symbols. Bulding SHARED library implies that by default.
https://cmake.org/cmake/help/v3.0/prop_tgt/POSITION_INDEPENDENT_CODE.html

Making heaptrack_unwind SHARED is an option here.